README.md

```diff
@@ -7,3 +7,522 @@ This repository contains submodules of all the repositories used for skywire dep
+* rf.skywire.skycoin.com/
+
+
+## Key generation
+
+Public/secret keypairs for all services have identical format can can be used interchangeably. There are multiple ways of potentially generating these, but for the sake of brevity the following command is suggested
+
+```
+skywire-cli config gen -n | head -n4 | tail -n2 | tee dmsgd-config.json
+grep -m1 "sk" dmsgd-config.json | cut -d '"' -f4 | tee -a dmsgd-config.json
+```
+
+those services which accept a secret key can then be run as follows
+```
+dmsg-discovery --sk $(tail -n1 dmsgd-config.json)
+```
+
+### Visor Config Bootstrap Endpoint
+
+The following endpoint is queried by the visor on config gen:
+https://conf.skywire.skycoin.com/
+
+This endpoint contains json which will become part of the visor's config
+```
+{
+  "dmsg_discovery": "http://dmsgd.skywire.skycoin.com",
+  "transport_discovery": "http://tpd.skywire.skycoin.com",
+  "address_resolver": "http://ar.skywire.skycoin.com",
+  "route_finder": "http://rf.skywire.skycoin.com",
+  "setup_nodes": [
+    "0324579f003e6b4048bae2def4365e634d8e0e3054a20fc7af49daf2a179658557"
+  ],
+  "uptime_tracker": "http://ut.skywire.skycoin.com",
+  "service_discovery": "http://sd.skycoin.com",
+  "stun_servers": [
+    "139.162.12.30:3478",
+    "170.187.228.181:3478",
+    "172.104.161.184:3478",
+    "170.187.231.137:3478",
+    "143.42.74.91:3478",
+    "170.187.225.78:3478",
+    "143.42.78.123:3478",
+    "139.162.12.244:3478"
+  ],
+  "dns_server": "1.1.1.1"
+}
+```
+
+This endpoint is used to avoid relying on hardcoded defaults for the production deployment endpoints in the visor. Without it, any changes here would require an updated version of skywire to manifest correct config default values
+
+A deployment of _all services as a set_ will require this for visors to query this endpoint during `config gen`
+
+### `dmsg-discovery`
+Run the Dmsg Discovery server
+```
+skywire-cli config gen -n | head -n4 | tail -n2 | tee dmsgd-config.json
+grep -m1 "sk" dmsgd-config.json | cut -d '"' -f4 | tee -a dmsgd-config.json
+dmsg-discovery --addr ":9090" --redis "redis://localhost:6379" --sk $(tail -n1 dmsgd-config.json)
+```
+
+### `dmsg-server`
+
+#### Configure `dmsg-server`
+Generate a config for the dmsg server
+```
+dmsg-server config gen -o dmsg-config.json
+```
+output
+```
+{
+	"public_key": "02a4072022716ee7fb8205edd4286dc73abe514ff8a7a74c6a27131c6efbe71876",
+	"secret_key": "62bcd18c5693ad1d2a24239f7303a84d6afb6966340fc70afe93579987cceb90",
+	"discovery": "http://dmsgd.skywire.skycoin.com",
+	"public_address": "127.0.0.1:8081",
+	"local_address": ":8081",
+	"health_endpoint_address": ":8082",
+	"log_level": "info",
+	"update_interval": 0,
+	"max_sessions": 2048
+}
+```
+
+The above "discovery" endpoint should be changed to match the endpoint of the dmsg discovery server referenced in the previous section.
+The default ports are shown.
+
+### Run `dmsg-server`
+
+Run the dmsg server (read config from file not working)
```

Comment:
Verify thats the case. Its working on prod.

Reply:
I would not have stated that or run the dmsg server in that would have read from the config. Neither the config existing at the default path `config.json` nor specifying the full path with `dmsg-server -c /path/to/config.json` would read that config.

I'm using develop. I will document this and file a ticket in the repo.